### PR TITLE
Do not select disabled items when select-all checkbox in List is clicked

### PR DESCRIPTION
**Pull request by nnatter**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Related issues/PRs | Builds upon #4992
| License | MIT

#### What's in this PR?

This PR adjusts the behaviour of the `List` container to only select **non-disabled** items if the`onAllSelectionChange` callback is fired by a `ListAdapter` (for example when the select-all checkbox of the `TableAdapter` is clicked). 

Furthermore, this PR fixes a bug that rendered the select-all checkbox of the `TableAdapter` with the wrong skin. Also, the select-all checkbox of the `TableAdapter` is now rendered as checked if all non-disabled rows are selected.

#### Why?

Because the goal of disabling items is that they cannot get selected.. 😉

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/tests/Table.test.js [hunk 1]
```diff
@@ -371,6 +390,59 @@ test('Clicking a checkbox should call onRowSelectionChange with the selection st
     expect(onChangeSpy).toHaveBeenCalledWith(rowIdOne, true);
 });
 
+test('Select-all checkbox should be checked if every non-disabled line is selected', () => {
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/tests/Table.test.js]: Any reason not to split this test into three different tests?

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/Header.js [hunk 2]
```diff
@@ -96,7 +96,7 @@ export default class Header extends React.PureComponent<Props> {
                     <Checkbox
                         checked={allSelected}
                         onChange={this.handleAllSelectionChange}
-                        skin={skin}
+                        skin={skin === 'dark' ? 'light' : 'dark'}
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/Header.js]: Why is this line necessary? As far as I see this will not change anything :thinking:

Comment (nnatter) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/Header.js]: it sets the skin of the checkbox to `light` if the skin of the header is `dark` and vice versa. this only affects tables that render the checkbox into the first cell (eg. the category list).

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/tests/Table.test.js (hunk 1 above)
Comment (nnatter) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Table/tests/Table.test.js]: done 🙂
